### PR TITLE
Adding publish and subscribe atomic client examples

### PR DESCRIPTION
**Pull request by embhorn**
New examples in `examples/pub-sub`

These can be used for quickly testing MQTT application or scripting.

By default, they connect to localhost. Specify a topic with `-n`
`./examples/pub-sub/mqtt-sub -n '#'`

`./examples/pub-sub/mqtt-pub -n test -m "Hello world!"`

Enable verbose example debug with `-d`

### CODE REVIEW

File: examples/pub-sub/mqtt-sub.c [hunk 1]
```diff
@@ -0,0 +1,655 @@
...
+        static void sig_handler(int signo)
+        {
+            if (signo == SIGINT) {
+                mStopRead = 1;
+            }
+        }
+    #endif
+
+#if defined(NO_MAIN_DRIVER)
+int mqttclient_main(int argc, char** argv)
```
Comment (lealem47) [inline examples/pub-sub/mqtt-sub.c:607]: Rename to avoid clashing names. `mqttSub_main()`. Also add that to mqtt-pub-sub.h header

**Review by lealem47 — CHANGES_REQUESTED** [review]
Works great. Should there be a new test script for the pub-sub clients?

**Comment by embhorn** [comment]
> Should there be a new test script for the pub-sub clients?

I hadn't considered this. These are really just subsets of the mqttclient example. They could be used for testing against the protocol.

**Review by lealem47 — DISMISSED** [review]

**Comment by embhorn** [comment]
Fixed executable names in .gitignore

**Review by dgarske — APPROVED** [review]
All looks good and tests out. I notice these aren't being used yet. Are you planning to add some additional scripts to test using these?

**Comment by embhorn** [comment]
> All looks good and tests out. I notice these aren't being used yet. Are you planning to add some additional scripts to test using these?

Yes, that is the intention, to better enable functional testing using these clients.

**Pull request merged**